flocker/control/_protocol.py

```diff
@@ -37,12 +37,37 @@
     Argument, Command, Integer, CommandLocator, AMP, Unicode, ListOf,
 )
 from twisted.internet.protocol import ServerFactory
+from twisted.internet.ssl import PrivateCertificate
 from twisted.application.internet import StreamServerEndpointService
+from twisted.protocols.tls import TLSMemoryBIOFactory
+from ..ca import ControlCredential, DEFAULT_CERTIFICATE_PATH
 
 from ._persistence import wire_encode, wire_decode
 from ._model import Deployment, NodeState, DeploymentState, NonManifestDatasets
 
 
+def tls_context_factory(path):
+    """
+    Create the security properties context factory for a ControlCredential
+    TLS connection using the control service certificate found in the
+    supplied path.
+
+    :param FilePath path: Absolute path to directory containing
+        control service certificate and private key files.
+
+    :return CertificateOptions: A context factory pre-loaded with the
+        options of the control service certificate.
+    """
+    if path is None:
```

Comment:
This (i.e. default certificate directory) is library code. Configuration, especially default configuration, should only be in application code, a thin layer on top of the library code. In this case that would probably be `flocker/control/script.py`.